### PR TITLE
FEAT-#5797: Add a QC method to reset to a positional index lazily.

### LINKED ISSUE #5797: Add a QC method to determine if index is positional
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is. What kind of performance improvements would you like to see with this new API?
When creating a modin.numpy array, we reset the index to be positional always. We should add a QC method to determine whether or not the index of an object is positional, and only replace it if it is not!

### PR DESCRIPTION
**Pull request by RehanSD**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5797 ? 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/core/storage_formats/pandas/query_compiler.py [hunk 1]
```diff
@@ -737,6 +737,16 @@ def _reset(df, *axis_lengths, partition_idx):
             )
         return new_self
 
+    def maybe_reset_index_to_positional(self):
+        if self.index.equals(pandas.RangeIndex(len(self.index))):
```
Comment (dchigarev) [inline modin/core/storage_formats/pandas/query_compiler.py:741]: Can we just add this condition to already existing `qc.reset_index` method?

File: modin/core/storage_formats/pandas/query_compiler.py [hunk 2]
```diff
@@ -737,6 +737,16 @@ def _reset(df, *axis_lengths, partition_idx):
...
+            return self
+        return self.reset_index(drop=True)
+
+    def maybe_reset_columns_to_positional(self):
```
Comment (dchigarev) [inline modin/core/storage_formats/pandas/query_compiler.py:745]: We could have also added the `axis` parameter to the existing `qc.reset_index` method in order not to produce new methods to already overloaded query compiler class. What do you think?

File: modin/core/storage_formats/base/query_compiler.py [hunk 3]
```diff
@@ -1150,6 +1150,29 @@ def reset_index(self, **kwargs):  # noqa: PR02
...
+        """
+        Reset the columns to a positional index, if it is not already.
+
+        Returns
+        -------
+        BaseQueryCompiler
+            QueryCompiler with appropriate columns.
+        """
+        self.columns = range(len(self.columns))
+        return self
```
Comment (vnlitvinov) [inline modin/core/storage_formats/base/query_compiler.py:1174]: this is kind of breaking the logic of always returning a new object...

**Review by vnlitvinov — COMMENTED** [review]
I think I'm on Dmitry's side here - it feels like we're adding two methods instead of two parameters to an existing `qc.reset_index` instead...

**Comment by RehanSD** [comment]
Actually this PR is no longer needed.

**Comment by RehanSD** [comment]
This PR is no longer necessary (https://github.com/modin-project/modin/issues/5797#issuecomment-1558832628 explains why), so I am closing it.

**Pull request closed without merging**